packages/react-form/src/hooks/field/field.ts

```diff
@@ -195,6 +195,45 @@ export function useField<Value = string>(
   return field as Field<Value>;
 }
 
+/**
+ * Converts a standard `Field<boolean>` into a `ChoiceField` that is compatible
+ * with `<Checkbox />` and `<RadioButton />` components in `@shopify/polaris`.
+ *
+ * For fields that are used by both a choice components and other components, it
+ * can be beneficial to retain the original `Field<boolean>` shape and convert
+ * the field on the fly for the choice component.
+ *
+ * ```typescript
+ * const enabled = useField(false);
+ * return (<Checkbox label="Enabled" {...asChoiceField(enabled)} />);
+ * ```
+ */
+export function asChoiceField({value: checked, ...fieldData}: Field<boolean>) {
+  return {
+    checked,
+    ...fieldData,
+  };
+}
+
+export type ChoiceField = ReturnType<typeof asChoiceField>;
+
+/**
+ * A simplification to `useField` that returns a `ChoiceField` by automatically
+ * converting the field using `asChoiceField` for direct use in choice
+ * components.
+ *
+ * ```typescript
+ * const enabled = useChoiceField(false);
+ * return (<Checkbox label="Enabled" {...enabled} />);
+ * ```
+ */
+export function useChoiceField(
```

Comment:
@TheMallen i moved the functions straight into this module. IMO it feels more natural to put them here.